packages/react-vapor/src/components/navigation/pagination/NavigationPaginationReducers.ts

```diff
@@ -37,6 +38,10 @@ export const paginationReducer = (
                 : state;
         case FilterActions.filterThrough:
             return s.contains(state.id, action.payload.id) ? {...state, pageNb: 0} : state;
+        case ListBoxActions.select: {
+            const [tableId] = s.words(action.payload.id, '--');

```

Comment:
I understand what you're doing here but I think it would be better to add an util inside `TableHOCUtils` to get the TableId from the predicateid, this way here you wouldn't need to know that the separator is `--`

Also, I think it's easier to understand with a split or a regex than with the words function